File: Loading Views/LoadingViews/Presentations Option 0 Pop Over/VCThatCallsPopOverViewController.m
//
//  VCThatCallsPopOverViewController.m
//  LoadingViews
//
//  Created by Luis Castillo on 12/30/15.
//  Copyright © 2015 DevBros. All rights reserved.
//

#import "VCThatCallsPopOverViewController.h"

#import "PopOverViewController.h"

@interface VCThatCallsPopOverViewController ()

@end

@implementation VCThatCallsPopOverViewController

@synthesize popOverVC;


    //MARK: View Loading
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}


    //MARK: Show Pop Over
- (IBAction)showMePopOver:(id)sender
{
    
    self.popOverVC = [[PopOverViewController alloc]initWithNibName:@"PopOverViewController" bundle:nil];
    
    /* updating popover view with exact frame (taking into consideration navbar as well) */
    if (self.navigationController.navigationBar != nil)
    {
        CGFloat navBarHeight = self.navigationController.navigationBar.frame.size.height;
        CGFloat calcYOrigin = self.view.bounds.origin.y + navBarHeight+1;
        CGFloat calcHeight = self.view.bounds.size.height - navBarHeight-1;
        
        CGRect popOverFrame = CGRectMake(self.view.bounds.origin.x,
                                         calcYOrigin,
                                         self.view.bounds.size.width,
                                         calcHeight);
        
        self.popOverVC.view.frame = popOverFrame;
    }
    else
    {
         self.popOverVC.view.frame = self.view.bounds;
    }
    
    UIImage * starWarsImage = [UIImage imageNamed:@"starWars.jpg"];
    
    [self.popOverVC showInView:self.view withImage:starWarsImage withMessage:@"May the Force be with You" animated:true];
    
}//eo-a

@end

File: Loading Views/LoadingViews/Presentations Option 0 Pop Over/VCThatCallsPopOverViewController.swift
//
//  VCThatCallsPopOverViewController.swift
//  LoadingViews
//
//  Created by Luis Castillo on 1/1/16.
//  Copyright © 2016 devbros. All rights reserved.
//

import UIKit
import QuartzCore

class VCThatCallsPopOverViewControllerSwift: UIViewController {

    var popOverVC:PopOverViewControllerSwift!
    
    //MARK: View Loading
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }//eom

    //MARK: Memory
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - Navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    
    //MARK: Show PopOver
    
    @IBAction func showPopOver(_ sender: AnyObject)
    {
        self.showPopView()
    }//eoa
   
    
    func showPopView()
    {
        self.popOverVC = PopOverViewControllerSwift(nibName: "PopOverViewController", bundle: nil)
        
        if navigationController != nil
        {
            let navBarHeight:CGFloat = self.navigationController!.navigationBar.frame.size.height
            let calcYOrigin:CGFloat = self.view.bounds.origin.y + navBarHeight + 1
            let calcHeight:CGFloat = self.view.bounds.size.height - navBarHeight - 1
            
            let popOverFrame = CGRect(x: self.view.bounds.origin.x,
                                      y: calcYOrigin,
                                      width: self.view.bounds.size.width,
                                      height: calcHeight)
            
            self.popOverVC.view.frame = popOverFrame
        }
        else
        {
            self.popOverVC.view.frame = self.view.bounds
        }
        
        
        self.popOverVC .showInView(self.view, animated: true)
    }//eom

}
